addons/web_editor/static/lib/odoo-editor/test/spec/format.test.js

```diff
@@ -398,41 +494,42 @@ describe('Format', () => {
             await testEditor(BasicEditor, {
                 contentBefore: `<p>${s(`ab[]cd`)}</p>`,
                 stepFunction: strikeThrough,
-                contentAfterEdit: `<p>${s(`ab`)}<span oe-zws-empty-inline="">\u200B[]</span>${s(`cd`)}</p>`,
-                contentAfter: `<p>${s(`ab`)}[]${s(`cd`)}</p>`,
+                contentAfterEdit: `<p>${s(`ab`)}${span(`[]\u200B`, 'first')}${s(`cd`)}</p>`,
+                contentAfter: `<p>${s(`ab[]cd`)}</p>`,
             });
         });
     });
+
     describe('underline + strikeThrough', () => {
         it('should get ready to write in strikeThrough without underline (underline was first)', async () => {
             await testEditor(BasicEditor, {
                 contentBefore: `<p>ab${u(s(`cd[]ef`))}</p>`,
                 stepFunction: underline,
-                contentAfterEdit: `<p>ab${u(s(`cd`))}${s(`\u200b[]`, 'last')}${u(s(`ef`))}</p>`,
-                contentAfter: `<p>ab${u(s(`cd`))}[]${u(s(`ef`))}</p>`,
+                contentAfterEdit: `<p>ab${u(s(`cd`))}${s(`[]\u200b`, 'last')}${u(s(`ef`))}</p>`,
+                contentAfter: `<p>ab${u(s(`cd[]ef`))}</p>`,
             });
         });
         it('should restore underline after removing it (collapsed, strikeThrough)', async () => {
             await testEditor(BasicEditor, {
                 contentBefore: `<p>ab${u(s(`cd`))}${s(`\u200b[]`, 'first')}${u(s(`ef`))}</p>`,
                 stepFunction: underline,
                 contentAfterEdit: `<p>ab${u(s(`cd`))}${s(u(`[]\u200b`, 'first'), 'first')}${u(s(`ef`))}</p>`,
-                contentAfter: `<p>ab${u(s(`cd`))}[]${u(s(`ef`))}</p>`,
+                contentAfter: `<p>ab${u(s(`cd[]ef`))}</p>`,
             });
         });
         it('should remove underline after restoring it after removing it (collapsed, strikeThrough)', async () => {
             await testEditor(BasicEditor, {
                 contentBefore: `<p>ab${u(s(`cd`))}${s(u(`[]\u200b`, 'first'))}${u(s(`ef`))}</p>`,
                 stepFunction: underline,
-                contentAfterEdit: `<p>ab${u(s(`cd`))}${s(`\u200b[]`, 'last')}${u(s(`ef`))}</p>`,
-                contentAfter: `<p>ab${u(s(`cd`))}[]${u(s(`ef`))}</p>`,
+                contentAfterEdit: `<p>ab${u(s(`cd`))}${s(`[]\u200b`, 'last')}${u(s(`ef`))}</p>`,
```

Comment:
Wait, the zws changed place ?

Reply:
The selection is slightly different. But it does not change the behavior.